rgwadmin/radosgw.go

```diff
@@ -0,0 +1,150 @@
+package rgwadmin
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	v4 "github.com/aws/aws-sdk-go/aws/signer/v4"
+)
+
+const (
+	authRegion = "default"
+	service    = "s3"
+)
+
+var (
+	errNoEndpoint  = errors.New("endpoint not set")
+	errNoAccessKey = errors.New("access key not set")
+	errNoSecretKey = errors.New("secret key not set")
+)
+
+//API struct for New Client
+type API struct {
+	AccessKey  string
+	SecretKey  string
+	Endpoint   string
+	HTTPClient *http.Client
+	Debug      bool
+}
+
+// New returns client for Ceph RGW
+func New(endpoint, accessKey, secretKey string, httpClient *http.Client) (*API, error) {
+	// validate endpoint
+	if endpoint == "" {
+		return nil, errNoEndpoint
+	}
+
+	// validate access key
+	if accessKey == "" {
+		return nil, errNoAccessKey
+	}
+
+	// validate secret key
+	if secretKey == "" {
+		return nil, errNoSecretKey
+	}
+
+	// set http client, if TLS is desired it will have to be passed with an http client
```

Comment:
If TLS is the only concern here it is better to pass tlscert as an agreement than the entire httpclient struct. And this API can create httpclient using that

Reply:
I was thinking about this, but they are probably other things that could be passed so the entire client looks safer to me so we will never have to revisit this in the futur.

Reply:
Okay. I am a bit worried because the user can also misuse this option. For example, they enable `InsecureSkipVerify` to bypass the TLS authentication.